aviary/interface/methods_for_level2.py

```diff
@@ -1436,20 +1455,82 @@ def add_linkage_constraint(self, phase_a, phase_b, var_a, var_b, connected,
+                    # and neither phase is ground roll (altitude isn't a state): we want altitude to be continous as well
+                    if ((phase1 in self.reserve_phases) == (phase2 in self.reserve_phases)) and \
+                            not ('groundroll' in (phase1, phase2)):
+                        states_to_link.append(Dynamic.Mission.ALTITUDE)
+                    # if either phase is rotation, we need to connect velocity and alpha
+                    if phase1 == 'rotation' or phase2 == 'rotation':
+                        states_to_link.append(Dynamic.Mission.VELOCITY)
+                        states_to_link.append('alpha')
+
+                    analytic1 = self.phase_info[phase1]['user_options']['analytic']
+                    analytic2 = self.phase_info[phase2]['user_options']['analytic']
+
+                    for state in states_to_link:
+                        # in initial guesses, all of the states, other than time use the same name
+                        temp_state = 'times' if state == 'time' else state
+                        initial_guesses1 = self.phase_info[phase1]['initial_guesses']
+                        initial_guesses2 = self.phase_info[phase2]['initial_guesses']
+
+                        # if a state is in the initial guesses, get the units of the initial guess
+                        units1, units2 = True, True
+                        if temp_state in initial_guesses1:
+                            units1 = initial_guesses1[temp_state][-1]
+                        if temp_state in initial_guesses2:
+                            units2 = initial_guesses2[temp_state][-1]
+                        else:
+                            units1 = units2 = True
+
+                        # if neither phase is analytic and the units of the initial guesses match we can use link_phases
+                        if not (analytic1 or analytic2) and (units1 == units2):
+                            if state == 'alpha':
+                                # alpha is always connected with a constraint
+                                connected = False
+                            else:
+                                # we can't use connected=True if we are running in parallel, but we usually want it
+                                connected = true_unless_mpi
+                            self.traj.link_phases(
+                                [phase1, phase2], [state], connected=connected)
+
+                        # if either phase is analytic or the units of the intial guesses don't match, we have to use a linkage_constraint
+                        else:
+                            # analytic phases use the prefix "initial" for time and distance, but not mass
+                            if analytic2:
+                                prefix = 'initial_'
+                            else:
+                                prefix = ''
+                            add_linkage_constraint(self, phase1, phase2,
```

Comment:
It seems like these calls to `add_linkage_constraint` will be made multiple times because we are under a loop over states to link.

Reply:
good point